typings/react-tracking.d.ts

```diff
@@ -7,8 +7,8 @@
 declare module "react-tracking" {
   import * as React from "react"
 
-  export interface TrackingProp {
-    trackEvent({}): any
+  export interface TrackingProp<P = {}> {
+    trackEvent<T = {}>({  }: Partial<P> & T): any
```

Comment:
This locks down what we can pass into `trackEvent`. The general flow (most of which the end-user won't touch, is like so: 

In the tracking context during overall app setup, pass our trackables into a (new) generic argument on `react-tracking`'s `TrackingProp`:
```
export interface TrackingContextProps {
  tracking?: TrackingProp<Trackables>
}
```
This constrains, in general, what we can pass into the function. 

Then later, in the rare occurrence when there's a one-off prop that doesn't belong in Trackables, the end-user can pass it in as a generic to `trackEvent`:

```
tracking.trackEvent<{ someRandomProp: string }>({
  ...
})
```

Reply:
Was coding a bit too late last night 🤦‍♂and returning to this it feels somewhat loopy.... I realized here that a slightly simpler way to do this would be to inject `trackEvent` rather than the `tracking` object, and then type things right in Reaction. 

It seems a bit windy to pass a generic into `TrackingProp` only to pass it down to `trackEvent` _and then_ further narrow the function with a generic specific to the function, only later -- and only sometimes. Will play with this a bit more. 

(This would also allow devs to call `trackEvent`, rather than `tracking.trackEvent`, which will simplify addressing the "it should still work" comment below.)